NebulaClient/MonoBehaviours/MultiplayerSession.cs

```diff
@@ -88,11 +99,24 @@ void Update()
         private void OnJoinSessionConfirmed(JoinSessionConfirmed packet)
         {
             PlayerManager.SetLocalPlayer(packet.LocalPlayerId);
+            if (statusBox != null)
+            {
+                statusBox.FadeOut();
+                statusBox = null;
+            }
         }
 
         private void OnRemotePlayerJoined(RemotePlayerJoined packet)
         {
             PlayerManager.AddRemotePlayer(packet.PlayerId);
+            statusBox = UIMessageBox.Show("A new player is joining!", "A new player is joining, please wait while they load in", null, 0);
+            GameMain.Pause();
+
+            if(PlayerManager.WeAreMainPlayer)
+            {
+                GameSave.SaveCurrentGame("MPSYNCSTATE");
+                Client.SendPacket(new InitialState(GameConfig.gameSaveFolder + "/MPSYNCSTATE.dsv"));
```

Comment:
We should use `GameSave.saveExt` instead of hardcoding the save file extension. If they ever change the extension to something else it will still work.